src/util/mpir_pmi.c

```diff
@@ -68,7 +69,14 @@ int MPIR_pmi_init(void)
     MPIR_ERR_CHKANDJUMP1(pmi_errno != PMI_SUCCESS, mpi_errno, MPI_ERR_OTHER,
                          "**pmi_kvs_get_value_length_max",
                          "**pmi_kvs_get_value_length_max %d", pmi_errno);
-
+    if (has_parent) {
+        pmi_parent_port = (char *) MPL_malloc(pmi_max_kvs_name_length, MPL_MEM_OTHER);
+        pmi_errno = PMI_KVS_Get(pmi_kvs_name, MPIDI_PARENT_PORT_KVSKEY, pmi_parent_port,
+                                MPIDI_MAX_KVS_VALUE_LEN);
+        MPIR_ERR_CHKANDJUMP1(pmi_errno != PMI_SUCCESS, mpi_errno, MPI_ERR_OTHER,
+                             "**pmi_kvs_get_parent_port",
+                             "**pmi_kvs_get_parent_port %d", pmi_errno);
+    }
```

Comment:
This seems move the spawn code back to early init. Why can't this code live together with the rest of the spawn code in `MPIR_Init_thread`? If we are concerned with the `MPIDI_PARENT_PORT_KVSKEY` namespace, that can be moved up.

Reply:
Actually, if we move this code into `MPIR_Init_thread` together with the rest of the spawn code, we don't even need keep `parent_port` as a global.

Reply:
That makes sense. Let me update this a bit.